examples/medplum-provider/src/components/messages/ThreadInbox.tsx

```diff
@@ -19,82 +19,107 @@ import {
-import { getReferenceString } from '@medplum/core';
+import { getReferenceString, parseSearchRequest } from '@medplum/core';
+import type { SearchRequest } from '@medplum/core';
 import { ChatList } from './ChatList';
 import { NewTopicDialog } from './NewTopicDialog';
 import { useThreadInbox } from '../../hooks/useThreadInbox';
 import classes from './ThreadInbox.module.css';
 import { useDisclosure } from '@mantine/hooks';
 import { showErrorNotification } from '../../utils/notifications';
 import cx from 'clsx';
+import { Link } from 'react-router';
 
 /**
  * ThreadInbox is a component that displays a list of threads and allows the user to select a thread to view.
  * @param query - The query to fetch all communications.
  * @param threadId - The id of the thread to select.
  * @param subject - The default subject when creating a new thread.
  * @param showPatientSummary - Whether to show the patient summary.
- * @param handleNewThread - A function to handle a new thread.
- * @param onSelectedItem - A function to handle the selected item.
+ * @param onNew - A function to handle a new thread.
+ * @param getThreadUri - A function to build thread URIs.
+ * @param onChange - A function to handle search changes.
+ * @param inProgressUri - The URI for in-progress threads.
+ * @param completedUri - The URI for completed threads.
  */
 
 interface ThreadInboxProps {
   query: string;
   threadId: string | undefined;
   subject?: Reference<Patient> | Patient | undefined;
   showPatientSummary?: boolean | undefined;
-  handleNewThread: (message: Communication) => void;
-  onSelectedItem: (topic: Communication) => string;
+  onNew: (message: Communication) => void;
+  getThreadUri: (topic: Communication) => string;
+  onChange: (search: SearchRequest) => void;
+  inProgressUri: string;
+  completedUri: string;
 }
 
 export function ThreadInbox(props: ThreadInboxProps): JSX.Element {
-  const { query, threadId, subject, showPatientSummary = false, handleNewThread, onSelectedItem } = props;
+  const {
+    query,
+    threadId,
+    subject,
+    showPatientSummary = false,
+    onNew,
+    getThreadUri,
+    onChange,
+    inProgressUri,
+    completedUri,
+  } = props;
 
   const [modalOpened, { open: openModal, close: closeModal }] = useDisclosure(false);
-  const [status, setStatus] = useState<Communication['status']>('in-progress');
-  const [currentPage, setCurrentPage] = useState<number>(1);
 
-  const itemsPerPage = 20;
+  const searchParams = useMemo(() => new URLSearchParams(query), [query]);
+  const itemsPerPage = Number.parseInt(searchParams.get('_count') || '20', 10);
+  const currentOffset = Number.parseInt(searchParams.get('_offset') || '0', 10);
+  const currentPage = Math.floor(currentOffset / itemsPerPage) + 1;
+  const status = (searchParams.get('status') as Communication['status']) || 'in-progress';
 
-  const queryWithStatus = useMemo(() => `${query}&status=${status}`, [query, status]);
+  const effectiveQuery = query;
```

Comment:
It looks like this is creating an additional variable name binding without any real benefit? The code below actually gets a little less readable IMO (as it isn't using the object shorthand like `useThreadInbox({ query, threadId })`).

Is this following some other convention or pattern I should learn more about?